packets.go

```diff
@@ -222,9 +222,9 @@ func (mc *mysqlConn) readHandshakePacket() (data []byte, plugin string, err erro
 	if mc.flags&clientProtocol41 == 0 {
 		return nil, "", ErrOldProtocol
 	}
-	if mc.flags&clientSSL == 0 && mc.cfg.tls != nil {
-		if mc.cfg.TLSConfig == "preferred" {
-			mc.cfg.tls = nil
+	if mc.flags&clientSSL == 0 && mc.cfg.TLS != nil {
+		if mc.cfg.AllowFallbackToNoTLS {
+			mc.cfg.TLS = nil
```

Comment:
Why delete `TLSConfig == "preferred"`?

Reply:
Ok. I got it. 

Merging this PR is beneficial for some users with old versions of MySQL. 🎉
 
LGTM.